### ISSUE TITLE
Running test_ttk_guionly logs "test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed" warnings

### ISSUE THREAD

**Issue by vstinner** [python/cpython#75689]
BPO | [31508](https://bugs.python.org/issue31508)
--- | :---
Nosy | @vstinner, @serhiy-storchaka, @csabella
PRs | <li>python/cpython#3651</li><li>python/cpython#3667</li><li>python/cpython#3719</li>

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2017-09-24.12:05:15.793>
created_at = <Date 2017-09-18.14:09:37.028>
labels = ['3.7', 'expert-tkinter']
title = 'Running test_ttk_guionly logs "test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed" warnings'
updated_at = <Date 2018-02-01.16:49:29.694>
user = 'https://github.com/vstinner'
```

bugs.python.org fields:
```python
activity = <Date 2018-02-01.16:49:29.694>
actor = 'serhiy.storchaka'
assignee = 'none'
closed = True
closed_date = <Date 2017-09-24.12:05:15.793>
closer = 'serhiy.storchaka'
components = ['Tkinter']
creation = <Date 2017-09-18.14:09:37.028>
creator = 'vstinner'
dependencies = []
files = []
hgrepos = []
issue_num = 31508
keywords = ['patch']
message_count = 12.0
messages = ['302454', '302456', '302458', '302466', '302576', '302598', '302600', '302849', '302853', '302855', '302856', '311450']
nosy_count = 4.0
nosy_names = ['vstinner', 'gpolo', 'serhiy.storchaka', 'cheryl.sabella']
pr_nums = ['3651', '3667', '3719']
priority = 'normal'
resolution = 'fixed'
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue31508'
versions = ['Python 3.7']
```

</p></details>

**Comment by vstinner** [comment]
0:33:11 load avg: 4.82 [ 44/405/1] test_ttk_guionly passed -- running: test_subprocess (504 sec)
/buildbot/buildarea/3.x.ware-gentoo-x86.refleak/build/Lib/tkinter/test/test_ttk/test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed
  'Deprecated API of Treeview.selection() should be removed')
beginning 6 repetitions
123456
/buildbot/buildarea/3.x.ware-gentoo-x86.refleak/build/Lib/tkinter/test/test_ttk/test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed
  'Deprecated API of Treeview.selection() should be removed')
./buildbot/buildarea/3.x.ware-gentoo-x86.refleak/build/Lib/tkinter/test/test_ttk/test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed
  'Deprecated API of Treeview.selection() should be removed')
./buildbot/buildarea/3.x.ware-gentoo-x86.refleak/build/Lib/tkinter/test/test_ttk/test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed
  'Deprecated API of Treeview.selection() should be removed')
./buildbot/buildarea/3.x.ware-gentoo-x86.refleak/build/Lib/tkinter/test/test_ttk/test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed
  'Deprecated API of Treeview.selection() should be removed')
./buildbot/buildarea/3.x.ware-gentoo-x86.refleak/build/Lib/tkinter/test/test_ttk/test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed
  'Deprecated API of Treeview.selection() should be removed')
./buildbot/buildarea/3.x.ware-gentoo-x86.refleak/build/Lib/tkinter/test/test_ttk/test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed
  'Deprecated API of Treeview.selection() should be removed')
.

**Comment by vstinner** [comment]
I saw these warnings on:
http://buildbot.python.org/all/builders/x86%20Gentoo%20Refleaks%203.x/builds/96/steps/test/logs/stdio

Extract of the pythoninfo:

tkinter.TCL_VERSION: 8.6
tkinter.TK_VERSION: 8.6
tkinter.info_patchlevel: 8.6.6

--

I can reproduce the warning locally:

test_selection (tkinter.test.test_ttk.test_widgets.TreeviewTest) ... /home/haypo/prog/python/master/Lib/tkinter/test/test_ttk/test_widgets.py:1562: UserWarning: Deprecated API of Treeview.selection() should be removed
  'Deprecated API of Treeview.selection() should be removed')
ok

**Comment by serhiy-storchaka** [comment]
This is a reminder to me to remove a deprecated API (see bpo-27319).

**Comment by vstinner** [comment]
> This is a reminder to me to remove a deprecated API (see bpo-27319).

Running tests should not log deprecation warnings.

Either remove the code, or make the warning quiet, no? :-)

**Comment by csabella** [comment]
Serhiy,

In tkinter \_\_init__.py, there's messages on trace_variable and other trace functions about adding a deprecation warning.  I didn't know if you intended to make those changes as well.

**Comment by serhiy-storchaka** [comment]
Thank you Cheryl. I'll open a separate issue.

**Comment by serhiy-storchaka** [comment]
Oh, I missed that the selection() method with its current behavior is documented in the module documentation. It isn't undocumented implementation detail. This raises a bar for removing higher.

I think it is better to defer removing to 3.8 or later. Instead we should document the deprecation.

**Comment by serhiy-storchaka** [comment]
New changeset 2fad10235460ac394cc8b869c41f47aba3d63594 by Serhiy Storchaka in branch 'master':
bpo-27319, bpo-31508: Document deprecation in Treeview.selection(). (bpo-3667)
https://github.com/python/cpython/commit/2fad10235460ac394cc8b869c41f47aba3d63594

**Comment by serhiy-storchaka** [comment]
New changeset e31eca45e548bf6f439d540f3751516acbc31689 by Serhiy Storchaka (Miss Islington (bot)) in branch '3.6':
[3.6] bpo-27319, bpo-31508: Document deprecation in Treeview.selection(). (GH-3667) (bpo-3719)
https://github.com/python/cpython/commit/e31eca45e548bf6f439d540f3751516acbc31689

**Comment by serhiy-storchaka** [comment]
Warnings are fixed in 3.7. They will reappear in 3.8.

**Comment by vstinner** [comment]
Thank you for the fix Serhiy!
